web/src/app/pages/create-survey/task-details/task-details.component.ts

```diff
@@ -110,6 +111,29 @@ export class TaskDetailsComponent {
       });
   }
 
+  onTaskDuplicate(index: number) {
+    this.dialogService
+      .openConfirmationDialog(
+        'Warning',
+        'Are you sure you wish to duplicate this task? Any associated data'
```

Comment:
If this just copies the task into a new one and doesn't delete any data, I don't think we need a warning here.